sprint_08

andriyanovme opened the pull request.

andriyanovme committed:
дошел до конца.

andriyanovme committed:
deselectRow добавил для выделения таблицы.

Nikitaaeee commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -9,6 +9,7 @@ import UIKit
 
 class ImagesListViewController: UIViewController {
     @IBOutlet private var tableView: UITableView!
+    
```
- Для улучшения производительности можно добавлять модификатор `final` к классам, от которых мы не будем наследоваться. (Как у тебя сделано в **ImagesListCell**, например) 
В таком случае комплятор будет быстрее находить адрес ячейки памяти при вызове функции.
Если интересно, тут можно подробнее почитать про то, что такое диспетчеризация и как она работает (довольно частая тема на собеседованиях): 
https://habr.com/ru/companies/simbirsoft/articles/673636/

- Для @IBOutlet принято добавлять модификатор `weak`
Вероятность retainCycle тут мала, но и нам ничего не мешает просто добавить `weak`

Nikitaaeee commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -20,33 +21,55 @@ class ImagesListViewController: UIViewController {
+        tableView.contentInset = UIEdgeInsets(top: 12, left: 0, bottom: 12, right: 0)
+    }
+    
+    func configCell(for cell: ImagesListCell, with indexPath: IndexPath) {
+        guard let image = UIImage(named: photosName[indexPath.row]) else {
+            return
+        }
+        
+        cell.cellImage.image = image
+        cell.dateLabel.text = dateFormatter.string(from: Date())
+        
+        let isLiked = indexPath.row % 2 == 0
+        let likeImage = isLiked ? UIImage(named: "Active") : UIImage(named: "No Active")
+        cell.likeButton.setImage(likeImage, for: .normal)
     }
```
Было бы круто вынести метод `configCell(...)` в класс ячейки.
В таком случае мы сможем "заприватить" свойства ячейки, так как обращение к ним уже будет происходить внутри **ImagesListCell**.
А также так будет архитектурно корректнее :)

Nikitaaeee commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -20,33 +21,55 @@ class ImagesListViewController: UIViewController {
+        return photosName.count
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: ImagesListCell.reuseIdentifier, for: indexPath) // 1
-        
-        guard let imageListCell = cell as? ImagesListCell else { // 2
+        guard let imageListCell = tableView.dequeueReusableCell(withIdentifier: ImagesListCell.reuseIdentifier, for: indexPath) as? ImagesListCell else {
             return UITableViewCell()
         }
         
-        configCell(for: imageListCell, with: indexPath) // 3
-        return imageListCell // 4
+        configCell(for: imageListCell, with: indexPath)
+        return imageListCell
     }
 }
 
 extension ImagesListViewController: UITableViewDelegate {
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        
+        tableView.deselectRow(at: indexPath, animated: true)
+    }
+    
```
Для отдельных `extensions`, которые реализуют функционал отдельных протоколов, удобно добавлять комментарии, например `// MARK: - UITableViewDataSource`
Это упростит организацию на Minimap (открывается через Shift+Control+Command+M) и облегчит поиск по коду.

Про организацию кода можно дополнительно посмотреть тут:
https://github.com/RedMadRobot/RMR-swift-style-guide#code-organization

andriyanovme committed:
1. Испраил у dateLabel отступ снизу и слева от края изображения.
2. Убрал выделение у ячейки (Selection - None).
3. Поменял цвет статус бара на белый.

andriyanovme committed:
Final class. //MARK. weak во ImagesListViewController

Nikitaaeee commented on ImageFeed/ImagesList/ImagesListViewController.swift:
```diff
@@ -19,6 +19,10 @@ class ImagesListViewController: UIViewController {
         return formatter
     }()
     
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        .lightContent
+    }
```
👍🏻👍🏻👍🏻
